### ISSUE TITLE
Extend move_base to accept pose from different message types

### ISSUE THREAD

**Issue by rnunziata** [ros-planning/navigation#231]
Extend move_base to accept odometry (pose) from additional message types
geometry_msgs/PoseWithCovarianceStamped 
geometry_msgs/PoseWithCovariance

Enhancement request

**Comment by mikeferguson** [comment]
@rnunziata would the covariance be used for anything? Or is this intended just as a convenience so you don't have to convert your PoseWithCovariance to a Pose?

**Comment by rnunziata** [comment]
Only to get around the need to reissue the message.
I see these messages as Pose with inheritance from Stamp and Covariance.
A better method would be to allow a function similar to remap that would allow a message to be recast a subclass. This will reduce unnecessary network traffic and complexity.

**Comment by DLu** [comment]
Did you want to have move_base subscribe to PoseWithCovariance? Or did you want to modify the move_base action?

**Comment by rnunziata** [comment]
There is some missunderstanding.  dwa_local_planner which is part of move_base(?) subscribes to odom (nav_msgs/Odometry). Requiring the need to reissue the odometry published by robot_pose_ekf/odom_combined (geometry_msgs/PoseWithCovarianceStamped). The odometry and localization are combined, yes? Maybe they should not be combined here rather then changing movebase. Is my understanding correct...my apologies if I miss understood usage here.

**Comment by rnunziata** [comment]
I have passed the covariance to move base but the robot seems to have trouble with the covariance values from ekf. I had forgotten that I had to override the covariance when I reposed the msg as nav_msg in my code. I  do not know why this is but it makes it a moot point at least for me now. I am still relatively new to ROS (sorry) and it is not always clear why somethings work and other like this do not. How does move base use covariance (not documented)?
